lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,60 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        attr_reader :identity
+
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+        VALID_USER_HEADER_KEYS = %w(account_number user username email is_org_admin)
+
+        def self.current
+          Thread.current[:attr_current_user] = new
```

Comment:
@syncrou 
should this be Thread.current[:attr_current_user] ||= new

How would this get reset?
Do we need a with_user method?

Reply:
Regarding the `with_user` method - https://github.com/ManageIQ/manageiq-api-common/pull/31#issuecomment-459059377 ( Yes )

Reply:
Yes to the other comment too, I'll get that change out.